pkg/bspatch/byte_add_reader.go

```diff
@@ -0,0 +1,59 @@
+package bspatch
+
+import (
+	"errors"
+	"fmt"
+	"io"
+)
+
+var ErrUnequalReads = errors.New("byteAdder.Read: did not read equal number of bytes from both readers")
+
+type byteAddReader struct {
+	r1   io.Reader
+	r2   io.Reader
+	addb []byte
+}
+
+func newByteAddReader(r1 io.Reader, r2 io.Reader) byteAddReader {
+	return byteAddReader{
+		r1:   r1,
+		r2:   r2,
+		addb: make([]byte, ReadBufferSize),
+	}
+}
+
+// Read adds the bytes of the two readers and writes to using p as scratch space.
+func (ba byteAddReader) Read(p []byte) (written int, err error) {
```

Comment:
Named return parameters are discouraged unless you really need them to be named (e.g. to modify the return value in a defer). In particular, naked `return`s reduce clarity.